### PR TITLE
GraphQL Non blocking support

### PR DESCRIPTION
**Pull request by phillip-kruger**
This PR pulls in a new release from SmallRye (https://github.com/smallrye/smallrye-graphql/releases/tag/1.5.0) and adds supports for Non blocking operations. The same rule applies as with RESTEasy Reactive. 

- By default, normal response objects will execute blocking, except if marked with `@NonBlocking`
- By default, Uni and CompletionStage response objects will execute non-blocking, except if marked with `@Blocking`

Breaking change: The ExecutionService has changed to not respond with a result, but rather a writer needs to be passed in. a default [JsonObjectResponseWriter](https://github.com/smallrye/smallrye-graphql/blob/main/server/implementation/src/main/java/io/smallrye/graphql/execution/JsonObjectResponseWriter.java) exist to get the same result as before. 

Because this is a major new feature, non-blocking can be turned off using a config: `quarkus.smallrye-graphql.nonblocking.enabled=false`. By default it will be enabled

Signed-off-by: Phillip Kruger <phillip.kruger@gmail.com>

### CODE REVIEW

**Comment by phillip-kruger** [comment]
@geoand - There are small changes on how the locale resolver work for GraphQL (something that came in a few weeks ago) - I moved the resolver to the graphql extension and are now getting the locale from the graphql context. This helps to make sure the locale are propagated.

**Comment by geoand** [comment]
Understood, thanks.

**Comment by jmartisk** [comment]
This currently makes the hibernate-validator a required dependency of the smallrye-graphql extension. To make it properly optional, we will probably need to:
- Change the dependencies in smallrye-graphql extension's poms to be `<optional>true</optional>`. I think we should also switch from depending on `hibernate-validator` correctly to depending on `io.quarkus:quarkus-hibernate-validator`. Note that the deployment pom already has a dependency on it, it's in the `test` scope! We need it to be optional, but present for testing. It seems that an optional dependency is always present during tests.
- In `SmallRyeGraphQLProcessor#additionalBean` we need to detect whether the `hibernate-validator` extension is present before we produce `SmallRyeGraphQLLocaleResolver` as a bean. If we produce it when the extension is not there, it will blow up. This means we will probably need to add a Capability for the hibernate-validator extension, to be able to do this check.

**Comment by phillip-kruger** [comment]
Thanks @jmartisk - Agree. I can make those changes now in this PR.

File: extensions/smallrye-graphql/runtime/src/main/java/io/quarkus/smallrye/graphql/runtime/SmallRyeGraphQLAbstractHandler.java
Comment (jmartisk) [inline extensions/smallrye-graphql/runtime/src/main/java/io/quarkus/smallrye/graphql/runtime/SmallRyeGraphQLAbstractHandler.java:43]: I don't like how you write into this non-volatile field again for each request, can't we set it once and make it final, the same as `currentIdentityAssociation` and `currentVertxRequest`?

File: extensions/smallrye-graphql-client/deployment/src/test/java/io/quarkus/smallrye/graphql/client/deployment/model/TestingGraphQLApi.java
Comment (jmartisk) [inline extensions/smallrye-graphql-client/deployment/src/test/java/io/quarkus/smallrye/graphql/client/deployment/model/TestingGraphQLApi.java:36]: What's the reason for this change? This is a GraphQLApi so it should preferably use only public APIs, which `CurrentVertxRequest` is, but `ContextHelper` isn't. Does this mean that injecting `CurrentVertxRequest` in this case no longer works??

Comment (phillip-kruger) [reply extensions/smallrye-graphql-client/deployment/src/test/java/io/quarkus/smallrye/graphql/client/deployment/model/TestingGraphQLApi.java:36]: I can remember :) I think in certain cases it does not work. Let me double check that...

Comment (jmartisk) [reply extensions/smallrye-graphql-client/deployment/src/test/java/io/quarkus/smallrye/graphql/client/deployment/model/TestingGraphQLApi.java:36]: This is something that should work IMO. I remember advising a user on StackOverflow that this is the only way to obtain the headers. How else would we tell users to do it?

Comment (phillip-kruger) [reply extensions/smallrye-graphql-client/deployment/src/test/java/io/quarkus/smallrye/graphql/client/deployment/model/TestingGraphQLApi.java:36]: Well, at this point all header is available in the context. (via unwrap)
We can add methods on SmallRye Context to expose them. Then basically inject SmallRyeContext and get Headers

Comment (phillip-kruger) [reply extensions/smallrye-graphql-client/deployment/src/test/java/io/quarkus/smallrye/graphql/client/deployment/model/TestingGraphQLApi.java]: OK I reverted back to the old test. Everything seems fine.

**Comment by phillip-kruger** [comment]
@jmartisk - I made all the changes as you requested. I think it's all fine now. MP TCK is passing on my localhost. Please review :)

**Comment by mschorsch** [comment]
Any chance that this will be part of 2.9?

**Comment by phillip-kruger** [comment]
> Any chance that this will be part of 2.9?

Not sure at this stage. Waiting for a green CI and PR approval. Might only be the next version, or allowing it here with a config option to enable it but default back to prev version

**Comment by phillip-kruger** [comment]
I have now added `quarkus.smallrye-graphql.nonblocking.enabled=true` that by default will be false. This allows a more soft release, so by default there is no change, but when enabled this experimental feature kicks in.

**Comment by cescoffier** [comment]
The CI failures look related.

**Comment by phillip-kruger** [comment]
@mschorsch - we missed the 2.9 release. Look out for the CR of 2.10

**Comment by mschorsch** [comment]
OK. Thanks.

**Comment by phillip-kruger** [comment]
@jmartisk @cescoffier - CI is green. Please review when you have a chance :) Thanks

**Comment by gsmet** [comment]
@jmartisk @cescoffier FWIW, this is in the way of my Jakarta work so if we can get it in soon, that would be awesome.

**Comment by phillip-kruger** [comment]
> I have now added `quarkus.smallrye-graphql.nonblocking.enabled=true` that by default will be false. This allows a more soft release, so by default there is no change, but when enabled this experimental feature kicks in.

This is now the other way around : non-blocking can be turned off using a config: `quarkus.smallrye-graphql.nonblocking.enabled=false.` By default it will be enabled

**Review by jmartisk — APPROVED** [review]
LGTM, very nice work!

**Review by cescoffier — APPROVED** [review]

**Pull request merged**
